dhis-2/dhis-api/src/main/java/org/hisp/dhis/program/ProgramInstanceStore.java

```diff
@@ -138,4 +139,26 @@
      * @param programInstance the ProgramInstance to delete.
      */
     void hardDelete( ProgramInstance programInstance );
+
+    /**
+     * Executes a query to fetch all {@see ProgramInstance} matching the
+     * Program/TrackedEntityInstance list and where Program Status is active.
+     *
+     * Resulting SQL query:
+     *
+     * <pre>{@code
+     *  select programinstanceid, programid, trackedentityinstanceid
+     *      from programinstance
+     *      where (programid = 726 and trackedentityinstanceid = 19 and status = 'ACTIVE')
+     *         or (programid = 726 and trackedentityinstanceid = 18 and status = 'ACTIVE')
+     *         or (programid = 726 and trackedentityinstanceid = 17 and status = 'ACTIVE')
+     * }</pre>
+     * 
+     * @param programTeiPair a List of Pair, where the left size is a {@see Program}
+     *        and the left side is a {@see TrackedEntityInstance}
+     * @return a List of {@see ProgramInstance}
+     */
+    List<ProgramInstance> getByProgramAndTrackedEntityInstance(
```

Comment:
Wouldn't it be better to have a more generic method where you can also send in ProgramStatus?